### PR TITLE
[Keyboard] add Ask55

### PR DESCRIPTION
**Pull request by keyboard-magpie**
## Description

Add the ask55 PCB designed by yiancar with qmk code contemporised by myself.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by spidey3** [comment]
LGTM, but your copyright comments seem to be out-of-date (2019), and don't indicate a GPL2+ compatible license.

Please update to add a GPL2+ compatible license header.
All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2022 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/ask55/info.json [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    },
+    "usb": {
+        "device_version": "0.0.1",
+        "pid": "0x3535",
+        "vid": "0x8968"
+    },
+    "layouts": {
+        "LAYOUT": {
+            "c_macro": true,
+            "filename": "keyboards/ask55/ask55.h",
```
Comment (zvecr) [inline keyboards/ask55/info.json:60]:
```suggestion
```

File: keyboards/ask55/ask55.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/* Copyright 2019 Yiancar / Keyboard-Magpie
```
Comment (zvecr) [inline keyboards/ask55/ask55.h:1]: This file should be removed, as matrix data is within info.json

File: keyboards/ask55/ask55.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/* Copyright 2019 Yiancar / Keyboard-Magpie
```
Comment (zvecr) [inline keyboards/ask55/ask55.c:1]: This file should be removed.

File: keyboards/ask55/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (zvecr) [inline keyboards/ask55/keymaps/default/keymap.c:49]:
```suggestion
```

File: keyboards/ask55/keymaps/iic/keymap.c [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (zvecr) [inline keyboards/ask55/keymaps/iic/keymap.c:49]:
```suggestion
```

File: keyboards/ask55/keymaps/m0118/keymap.c [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (zvecr) [inline keyboards/ask55/keymaps/m0118/keymap.c:49]:
```suggestion
```

File: keyboards/ask55/keymaps/mac-portable/keymap.c [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (zvecr) [inline keyboards/ask55/keymaps/mac-portable/keymap.c:49]:
```suggestion
```

File: keyboards/ask55/keymaps/next/keymap.c [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (zvecr) [inline keyboards/ask55/keymaps/next/keymap.c:49]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Also, `rules.mk` should match `data/templates/keyboard/rules.mk`

File: keyboards/ask55/info.json [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": true,
+        "sleep_led": false,
+        "unicode": false
+    },
+    "keyboard_folder": "ask55",
```
Comment (zvecr) [inline keyboards/ask55/info.json:23]:
```suggestion
```

File: keyboards/ask55/info.json [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+    "matrix_pins": {
+        "cols": ["E6", "D0", "D2", "D3", "D5", "D4", "D6", "D7", "B4", "B5", "B6", "C6", "C7", "F7"],
+        "rows": ["B0", "B1", "D1", "F6", "F5"]
+    },
+    "mouse_key": {
+        "enabled": true
+    },
```
Comment (zvecr) [inline keyboards/ask55/info.json:30]:
```suggestion
```

**Review by zvecr — COMMENTED** [review]
Sorry, missed a few.

File: keyboards/ask55/readme.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
+# ASK55
+
+The ASK55 is a 14.5u, Alps-compatible PCB available from Paradigm.  
+
+Keyboard Maintainer: [yiancar](https://github.com/yiancar) / [keyboard-magpie](https://github.com/keyboard-magpie)
+Hardware Supported: Paradigm ASK55 PCB  
+Hardware Availability: Paradigm GB
```
Comment (fauxpark) [inline keyboards/ask55/readme.md:7]:
```suggestion
* Keyboard Maintainer: [yiancar](https://github.com/yiancar) / [keyboard-magpie](https://github.com/keyboard-magpie)
* Hardware Supported: Paradigm ASK55 PCB
* Hardware Availability: Paradigm GB
```

File: keyboards/ask55/keymaps/mac-portable/readme.md [hunk 12]
```diff
@@ -0,0 +1,3 @@
+# The Mac Portable keymap for the ASK55
+
+![Mac Portable](https://paradigm.parts/images/teasers/ASK55/references/mac-portable.jpg)
```
Comment (fauxpark) [inline keyboards/ask55/keymaps/mac-portable/readme.md:3]: Dead link.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ask55/keymaps/mac-portable/readme.md (hunk 12 above)
Comment (keyboard-magpie) [reply keyboards/ask55/keymaps/mac-portable/readme.md]: Sorry, missed that one. Fixed.

**Review by fauxpark — APPROVED** [review]

File: keyboards/ask55/info.json [hunk 13]
```diff
@@ -0,0 +1,125 @@
+{
+    "manufacturer": "Yiancar-Designs",
+    "keyboard_name": "ASK55",
+    "maintainer": "qmk",
+    "processor": "atmega32u4",
+    "bootloader": "caterina",
```
Comment (drashna) [inline keyboards/ask55/info.json:6]:
```suggestion
    "development_board": "promicro",
```

File: keyboards/ask55/info.json [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+        "sleep_led": false,
+        "unicode": false
+    },
+    "matrix_pins": {
+        "cols": ["E6", "D0", "D2", "D3", "D5", "D4", "D6", "D7", "B4", "B5", "B6", "C6", "C7", "F7"],
+        "rows": ["B0", "B1", "D1", "F6", "F5"]
+    },
+    "rgblight": {
+        "animations": {
+            "all": false,
```
Comment (drashna) [inline keyboards/ask55/info.json:29]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ask55/config.h [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+/*== or choose animations ==*/
+  #define RGBLIGHT_EFFECT_BREATHING
+  #define RGBLIGHT_EFFECT_RAINBOW_MOOD
+  #define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+  #define RGBLIGHT_EFFECT_SNAKE
+  #define RGBLIGHT_EFFECT_KNIGHT
+  #define RGBLIGHT_EFFECT_CHRISTMAS
+  #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+  #define RGBLIGHT_EFFECT_RGB_TEST
+  #define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline keyboards/ask55/config.h:33]: These are in the info.json.  Don't need them both. 
```suggestion
```

File: keyboards/ask55/info.json [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+        "animations": {
+            "alternating": false,
+            "breathing": false,
+            "christmas": false,
+            "knight": false,
+            "rainbow_mood": false,
+            "rainbow_swirl": false,
+            "rgb_test": false,
+            "snake": false,
+            "static_gradient": false
```
Comment (drashna) [inline keyboards/ask55/info.json:36]:
```suggestion
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true
```

File: keyboards/ask55/info.json [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+        },
+        "brightness_steps": 8,
+        "hue_steps": 8,
+        "led_count": 8,
+        "max_brightness": 255,
+        "pin": "F4",
+        "saturation_steps": 8,
+        "sleep": false
```
Comment (drashna) [inline keyboards/ask55/info.json:44]:
```suggestion
        "sleep": true
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**